### PR TITLE
web/satellite: pagination component implemented

### PR DESCRIPTION
**Pull request by NikolaiYurchenko**
What: 
Common pagination component for satellite 
Why:

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: web/satellite/src/components/common/PagesBlock.vue [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+<script lang="ts">
+    import { Component, Vue, Prop } from 'vue-property-decorator';
+    import { Page } from '@/types/pagination';
+
+    @Component
+    export default class PaginationArea extends Vue {
+        @Prop({default: []})
+        public readonly pages: Page[];
+        @Prop({default: () => false})
+        public readonly checkSelected: (pageIndex: number) => boolean;
```
Comment (crawter) [inline web/satellite/src/components/common/PagesBlock.vue]: lets create callback type

File: web/satellite/src/components/common/Pagination.vue [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+        public isLoading = false;
+        public pagesArray: Page[] = [];
+        public firstBlockPages: Page[] = [];
+        public middleBlockPages: Page[] = [];
+        public lastBlockPages: Page[] = [];
+
+        @Prop({default: 0})
+        private readonly totalPageCount: number;
+        @Prop({default: () => { return new Promise(() => false); }})
+        private readonly onPageClickCallback: (index: number) => Promise<any>;
```
Comment (crawter) [inline web/satellite/src/components/common/Pagination.vue]: lets create callback type

File: web/satellite/tests/unit/common/PagesBlock.spec.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+describe('Pagination.vue', () => {
+    it('renders correctly without props', () => {
+        const wrapper = shallowMount(PagesBlock);
+
+        expect(wrapper).toMatchSnapshot();
+    });
+
+    it('renders correctly with props', () => {
+        const callbackSpy = sinon.spy();
+        let pagesArray: Page[] = [];
```
Comment (crawter) [inline web/satellite/tests/unit/common/PagesBlock.spec.ts]: const?

File: web/satellite/src/components/common/Pagination.vue [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+                return;
+            }
+
+            if (this.isCurrentInFirstBlock()) {
+                this.setBlocksIfCurrentInFirstBlock();
+
+                return;
+            }
+
+            if (!this.isCurrentInFirstBlock() &&  !this.isCurrentInLastBlock()) {
```
Comment (crawter) [inline web/satellite/src/components/common/Pagination.vue]: extra space

File: web/satellite/src/components/common/PagesBlock.vue [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+            :key="page.index">{{page.index}}</span>
+    </div>
+</template>
+
+<script lang="ts">
+    import { Component, Vue, Prop } from 'vue-property-decorator';
+    import { Page } from '@/types/pagination';
+
+    @Component
+    export default class PaginationArea extends Vue {
```
Comment (rikysya) [inline web/satellite/src/components/common/PagesBlock.vue]: Duplicate class name, you probably meant `PagesBlock`

**Pull request merged**